ggshield/core/cache.py

```diff
@@ -44,13 +45,20 @@ def load_cache(self) -> bool:
 
     def update_cache(self, **kwargs: Any) -> None:
         if SECRETS_CACHE_KEY in kwargs:
-            self.last_found_secrets = kwargs.pop(SECRETS_CACHE_KEY)
+            self.last_found_secrets = [
+                IgnoredMatchSchema().load(data=secret)
```

Comment:
Here the code is going to create an `IgnoredMatchSchema` instance for each secret. It would be more efficient to do something like this:
```python
   schema = IgnoredMatchSchema()
   self.last_found_secrets = [schema.load(data=secret)
       for secret in kwargs.pop(SECRETS_CACHE_KEY)]
```